src/main/java/ch/njol/skript/lang/Expression.java

```diff
@@ -120,6 +120,20 @@ default Optional<T> getOptionalSingle(Event event) {
 	 */
 	boolean isSingle();
 
+	/**
+	 * Whether there's a possibility this could return a single value.
+	 * Designed for expressions that may return more than one value, but are equally appropriate to use where
+	 * only singular values are accepted, in which case a single value (out of all available values) will be returned.
+	 * An example would be functions that return results based on their inputs.
+	 * Ideally, this will return {@link #isSingle()} based on its known inputs at initialisation, but for some syntax
+	 * this may not be known (or a syntax may be intentionally vague in its permissible returns).
+	 * @return Whether this can be used by single changers
+	 * @see #getSingle(Event)
+	 */
+	default boolean permitSingle() {
```

Comment:
I wonder if something like `mayBeSingle` would be a better name.

Reply:
This was what I originally had, but I was concerned that the intended purpose isn't just for things that *could* be single but also for stuff that just wants to allow single use even if it isn't; a kind of "you can discard all other values and just have one if you like" thing.

Reply:
I like canBeSingle